ROOT does not compile on musl libc

I'm trying to compile ROOT v6.24/06 on Void Linux with musl libc, however the build fails with the following error message:
```cc
/builddir/root-6.24.06/core/clib/src/mmapsup.c:47:15: error: conflicting types for 'getpagesize'
   47 | extern size_t getpagesize PARAMS ((void));
      |               ^~~~~~~~~~~
In file included from /builddir/root-6.24.06/core/clib/res/mmprivate.h:56,
                 from /builddir/root-6.24.06/core/clib/src/mmapsup.c:26:
/usr/include/unistd.h:163:5: note: previous declaration of 'getpagesize' was here
  163 | int getpagesize(void);
      |     ^~~~~~~~~~~
```

The relevant part of `mmapsup.c` is:
```c
#if defined(R__LINUX) && !defined(R__GLIBC) && !defined(__CYGWIN__) \
   || (defined(__CYGWIN__) && (CYGWIN_VERSION_API_MAJOR > 0 || CYGWIN_VERSION_API_MINOR < 213))
extern size_t getpagesize PARAMS ((void));
#else
extern int getpagesize PARAMS ((void));
#endif
```

My understanding is that as musl doesn't define `R__GLIBC`, the condition resolves in favor of declaring `getpagesize` return value as `size_t`. However, this conflicts with the declaration present in `unistd.h` (which is included several lines above this), which declares it as `int` (in accordance with the POSIX standard).

The same code snippet also causes error during `core/clib/src/mvalloc.c` compilation.

It's not entirely clear to me why should `getpagesize` be declared in the first place, as it is declared in the included `unistd.h` file - my guess is for some kind of compatibility, although POSIX dropped it twenty years ago and Cygwin seems to have implemented it at least four years ago, likely more, I am not sure how this works on Windows - let alone as `size_t`. Thus my suggestion is to remove the above code block from both of the mentioned files entirely, which seems to fix the problem for me (at least until the compilation fails on another problem, which I haven't analyzed yet; and I'm still able to compile for glibc too) and my question is: would it break anything crucial (as I mentioned, `getpagesize` has been deprecated for more than twenty years now (at least on UNIX-like systems), so keeping support for some potentially obscure platforms which still expect it with `size_t` return value may not be of high importance (but again, I do not know what the situation is on Windows))? And if so, what is the right way to solve this?

Here's a [link](https://root-forum.cern.ch/t/compiling-error-conflicting-types-alpine-linux/28193) to a ROOT forum thread which describes the same problem (but does not provide any more information than is stated above).

Another compilation failure is caused by this error:
```gcc
[1/1065] Building CXX object core/base/CMakeFiles/Base.dir/src/TString.cxx.o                                                                                                                   
FAILED: core/base/CMakeFiles/Base.dir/src/TString.cxx.o                                                                                                                                        
/usr/bin/g++  -I/builddir/root-6.24.06/core/base/inc -I/builddir/root-6.24.06/build/builtins/pcre/PCRE-prefix/src/PCRE-build -I/builddir/root-6.24.06/core/base/res -I/builddir/root-6.24.06/co
re/foundation/res -I/builddir/root-6.24.06/core/clib/inc -I/builddir/root-6.24.06/core/meta/inc -I/builddir/root-6.24.06/core/cont/inc -I/builddir/root-6.24.06/core/thread/inc -I/builddir/roo
t-6.24.06/core/gui/inc -I/builddir/root-6.24.06/core/foundation/inc -I/builddir/root-6.24.06/build/ginclude -DNDEBUG -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -fdiagnost
ics-color=always -std=c++14 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread  -DNDEBUG -fPIC -std=c++14 -MD -MT core/base/CMak
eFiles/Base.dir/src/TString.cxx.o -MF core/base/CMakeFiles/Base.dir/src/TString.cxx.o.d -o core/base/CMakeFiles/Base.dir/src/TString.cxx.o -c /builddir/root-6.24.06/core/base/src/TString.cxx 
In file included from /builddir/root-6.24.06/core/base/src/TString.cxx:44:                                                                                                                     
/builddir/root-6.24.06/core/base/src/TString.cxx: In member function 'void TString::FormImp(const char*, __va_list_tag*)':                                                                     
/builddir/root-6.24.06/core/base/inc/Varargs.h:48:35: error: '__va_copy' was not declared in this scope; did you mean 'va_copy'?                                                                  48 | #     define R__VA_COPY(to, from) __va_copy((to), (from))                                                                                                                              
      |                                   ^~~~~~~~~                                                                                                                                            
/builddir/root-6.24.06/core/base/src/TString.cxx:2269:4: note: in expansion of macro 'R__VA_COPY'
 2269 |    R__VA_COPY(sap, ap);                                                                                                                                                                
      |    ^~~~~~~~~~                                                                                                                                                                          
/builddir/root-6.24.06/core/base/src/TString.cxx: In function 'char* SlowFormat(const char*, __va_list_tag*, int)':
/builddir/root-6.24.06/core/base/inc/Varargs.h:48:35: error: '__va_copy' was not declared in this scope; did you mean 'va_copy'?                                                               
   48 | #     define R__VA_COPY(to, from) __va_copy((to), (from))                         
      |                                   ^~~~~~~~~                                         
/builddir/root-6.24.06/core/base/src/TString.cxx:2366:4: note: in expansion of macro 'R__VA_COPY'
 2366 |    R__VA_COPY(sap, ap);
      |    ^~~~~~~~~~
/builddir/root-6.24.06/core/base/src/TString.cxx: In function 'char* Format(const char*, __va_list_tag*)':
/builddir/root-6.24.06/core/base/inc/Varargs.h:48:35: error: '__va_copy' was not declared in this scope; did you mean 'va_copy'?
   48 | #     define R__VA_COPY(to, from) __va_copy((to), (from))
      |                                   ^~~~~~~~~
/builddir/root-6.24.06/core/base/src/TString.cxx:2415:4: note: in expansion of macro 'R__VA_COPY'
 2415 |    R__VA_COPY(sap, ap);
      |    ^~~~~~~~~~
```
The relevant code is
```c
#if !defined(R__VA_COPY)
#  if defined(__GNUC__) && !defined(__FreeBSD__)
#     define R__VA_COPY(to, from) __va_copy((to), (from))
#  elif defined(__va_copy)
#     define R__VA_COPY(to, from) __va_copy((to), (from))
#  elif defined(va_copy)
#     define R__VA_COPY(to, from) va_copy((to), (from))
#  elif defined(_WIN32) && _MSC_VER < 1310
#     define R__VA_COPY(to, from) (*(to) = *(from))
#  else
#     define R__VA_COPY(to, from) ((to) = (from))
#  endif
#endif
```
I am actually not sure if this could be a bug in `gcc`, because [apparently](https://www.gnu.org/software/libc/manual/html_node/Argument-Macros.html) `__va_copy` should be defined by the compiler and thus should not(?) be affected by libc.
> These macros are no longer provided by the GNU C Library, but rather by the compiler. 

Anyway, is there any reason why the above code block could not be rewritten as
```c
#if !defined(R__VA_COPY)
#  if defined(va_copy) /* checking this first as va_copy is the one described by the C99 standard */
#     define R__VA_COPY(to, from) va_copy((to), (from))
#  elif defined(__va_copy)
#     define R__VA_COPY(to, from) __va_copy((to), (from))
#  elif defined(_WIN32) && _MSC_VER < 1310
#     define R__VA_COPY(to, from) (*(to) = *(from))
#  else
#     define R__VA_COPY(to, from) ((to) = (from))
#  endif
#endif
```
?

```cc
FAILED: core/unix/CMakeFiles/Unix.dir/src/TUnixSystem.cxx.o                                                                                                                           [87/1897]
/usr/bin/g++  -I/builddir/root-6.24.06/core/unix/inc -I/builddir/root-6.24.06/core/clib/res -I/builddir/root-6.24.06/core/foundation/res -I/builddir/root-6.24.06/core/base/inc -I/builddir/roo
t-6.24.06/core/clib/inc -I/builddir/root-6.24.06/core/meta/inc -I/builddir/root-6.24.06/core/cont/inc -I/builddir/root-6.24.06/core/thread/inc -I/builddir/root-6.24.06/core/textinput/inc -I/b
uilddir/root-6.24.06/core/gui/inc -I/builddir/root-6.24.06/core/foundation/inc -I/builddir/root-6.24.06/build/ginclude -DNDEBUG -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2
 -fdiagnostics-color=always -std=c++14 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread  -DNDEBUG -fPIC -std=c++14 -MD -MT cor
e/unix/CMakeFiles/Unix.dir/src/TUnixSystem.cxx.o -MF core/unix/CMakeFiles/Unix.dir/src/TUnixSystem.cxx.o.d -o core/unix/CMakeFiles/Unix.dir/src/TUnixSystem.cxx.o -c /builddir/root-6.24.06/cor
e/unix/src/TUnixSystem.cxx                                                                     
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx: In member function 'virtual TInetAddress TUnixSystem::GetSockName(int)':
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3085:33: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3085 |    if (getsockname(sock, &addr, &len) == -1) {                         
      |                                 ^~~~                                                                                                                                                   
      |                                 |                                                      
      |                                 int*                                                   
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:                                                                                                                
/usr/include/sys/socket.h:338:52: note:   initializing argument 3 of 'int getsockname(int, sockaddr*, socklen_t*)'
  338 | int getsockname (int, struct sockaddr *__restrict, socklen_t *__restrict);
      |                                                    ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx: In member function 'virtual TInetAddress TUnixSystem::GetPeerName(int)':
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3121:33: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3121 |    if (getpeername(sock, &addr, &len) == -1) {                                                                                                                                         
      |                                 ^~~~                                                   
      |                                 |                                                      
      |                                 int*                                                                                                                                                   
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:           
/usr/include/sys/socket.h:339:52: note:   initializing argument 3 of 'int getpeername(int, sockaddr*, socklen_t*)'
  339 | int getpeername (int, struct sockaddr *__restrict, socklen_t *__restrict);
      |                                                    ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx: In member function 'virtual int TUnixSystem::GetSockOpt(int, int, int*)':
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3495:63: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3495 |       if (getsockopt(sock, SOL_SOCKET, SO_SNDBUF, (char*)val, &optlen) == -1) {
      |                                                               ^~~~~~~ 
      |                                                               |       
      |                                                               int*          
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);                                                                                                               
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3501:63: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3501 |       if (getsockopt(sock, SOL_SOCKET, SO_RCVBUF, (char*)val, &optlen) == -1) {
      |                                                               ^~~~~~~
      |                                                               |
      |                                                               int*
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3507:66: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3507 |       if (getsockopt(sock, SOL_SOCKET, SO_OOBINLINE, (char*)val, &optlen) == -1) {
      |                                                                  ^~~~~~~
      |                                                                  |
      |                                                                  int*
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3513:66: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3513 |       if (getsockopt(sock, SOL_SOCKET, SO_KEEPALIVE, (char*)val, &optlen) == -1) {
      |                                                                  ^~~~~~~
      |                                                                  |
      |                                                                  int*
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3519:66: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3519 |       if (getsockopt(sock, SOL_SOCKET, SO_REUSEADDR, (char*)val, &optlen) == -1) {
      |                                                                  ^~~~~~~
      |                                                                  |
      |                                                                  int*
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
/builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:3525:66: error: invalid conversion from 'int*' to 'socklen_t*' {aka 'unsigned int*'} [-fpermissive]
 3525 |       if (getsockopt(sock, IPPROTO_TCP, TCP_NODELAY, (char*)val, &optlen) == -1) {
      |                                                                  ^~~~~~~
      |                                                                  |
      |                                                                  int*
In file included from /builddir/root-6.24.06/core/unix/src/TUnixSystem.cxx:101:
/usr/include/sys/socket.h:348:50: note:   initializing argument 5 of 'int getsockopt(int, int, int, void*, socklen_t*)'
  348 | int getsockopt (int, int, int, void *__restrict, socklen_t *__restrict);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
```
is the next error.

The problem stems from the fact, that the condition for using `socklen_t` is (among others) glibc:
```c
#if (defined(R__AIX) && !defined(_AIX43)) || \
    (defined(R__SUNGCC3) && !defined(__arch64__))
#   define USE_SIZE_T
#elif defined(R__GLIBC) || defined(R__FBSD) || \
      (defined(R__SUNGCC3) && defined(__arch64__)) || \
      defined(R__OBSD) || defined(MAC_OS_X_VERSION_10_4) || \
      (defined(R__AIX) && defined(_AIX43)) || \
      (defined(R__SOLARIS) && defined(_SOCKLEN_T))
#   define USE_SOCKLEN_T
#endif
```
[Here](https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xsh_chap02.html#tag_22_02_10_06) is some context:
> All socklen_t types were originally (in BSD UNIX) of type int. During the development of POSIX.1-2017, it was decided to change all buffer lengths to size_t, which appears at face value to make sense. When dual mode 32/64-bit systems came along, this choice unnecessarily complicated system interfaces because size_t (with long) was a different size under ILP32 and LP64 models. Reverting to int would have happened except that some implementations had already shipped 64-bit-only interfaces. The compromise was a type which could be defined to be any size by the implementation: socklen_t.

I am not sure how to approach this, because the standardization of `socklen_t` seems fairly recent, but in the long run, avoiding this whole types dance altogether and simply using explicit `socklen_t` seems to make most sense and will make for a simpler code. I will appreciate any comments, especially regarding any compatibility issues this might cause. Else I will add this to the PR.

--- a/core/unix/src/TUnixSystem.cxx
+++ b/core/unix/src/TUnixSystem.cxx
@@ -148,17 +148,6 @@
 #   define UTMP_NO_ADDR
 #endif
 
-#if (defined(R__AIX) && !defined(_AIX43)) || \
-    (defined(R__SUNGCC3) && !defined(__arch64__))
-#   define USE_SIZE_T
-#elif defined(R__GLIBC) || defined(R__FBSD) || \
-      (defined(R__SUNGCC3) && defined(__arch64__)) || \
-      defined(R__OBSD) || defined(MAC_OS_X_VERSION_10_4) || \
-      (defined(R__AIX) && defined(_AIX43)) || \
-      (defined(R__SOLARIS) && defined(_SOCKLEN_T))
-#   define USE_SOCKLEN_T
-#endif
-
 #if defined(R__LYNXOS)
 extern "C" {
    extern int putenv(const char *);
@@ -3073,13 +3062,7 @@ TInetAddress TUnixSystem::GetHostByName(const char *hostname)
 TInetAddress TUnixSystem::GetSockName(int sock)
 {
    struct sockaddr addr;
-#if defined(USE_SIZE_T)
-   size_t len = sizeof(addr);
-#elif defined(USE_SOCKLEN_T)
    socklen_t len = sizeof(addr);
-#else
-   int len = sizeof(addr);
-#endif
 
    TInetAddress ia;
    if (getsockname(sock, &addr, &len) == -1) {
@@ -3109,13 +3092,7 @@ TInetAddress TUnixSystem::GetSockName(int sock)
 TInetAddress TUnixSystem::GetPeerName(int sock)
 {
    struct sockaddr addr;
-#if defined(USE_SIZE_T)
-   size_t len = sizeof(addr);
-#elif defined(USE_SOCKLEN_T)
    socklen_t len = sizeof(addr);
-#else
-   int len = sizeof(addr);
-#endif
 
    TInetAddress ia;
    if (getpeername(sock, &addr, &len) == -1) {
@@ -3482,13 +3459,7 @@ int TUnixSystem::GetSockOpt(int sock, int opt, int *val)
 {
    if (sock < 0) return -1;
 
-#if defined(USE_SOCKLEN_T) || defined(_AIX43)
    socklen_t optlen = sizeof(*val);
-#elif defined(USE_SIZE_T)
-   size_t optlen = sizeof(*val);
-#else
-   int optlen = sizeof(*val);
-#endif
 
    switch (opt) {
    case kSendBuffer:

--- a/interpreter/cling/lib/Interpreter/Interpreter.cpp
+++ b/interpreter/cling/lib/Interpreter/Interpreter.cpp
@@ -453,7 +453,7 @@ namespace cling {
 
     // Intercept all atexit calls, as the Interpreter and functions will be long
     // gone when the -native- versions invoke them.
-#if defined(__linux__)
+#if defined(__GLIBC__)
     const char* LinkageCxx = "extern \"C++\"";
     const char* Attr = LangOpts.CPlusPlus ? " throw () " : "";
 #else

--- a/io/io/src/TMapFile.cxx
+++ b/io/io/src/TMapFile.cxx
@@ -107,9 +107,7 @@ robust Streamer mechanism I opted for 3).
 #include <sys/types.h>
 #include <sys/ipc.h>
 #include <sys/sem.h>
-#if defined(R__HPUX) || \
-    defined (R__SOLARIS) || defined(R__AIX) || defined(R__HIUX) || \
-    __GLIBC_MINOR__ > 0
+#ifndef WIN32
 union semun {
    int val;                      // value for SETVAL
    struct semid_ds *buf;         // buffer for IPC_STAT & IPC_SET

--- a/math/mathcore/inc/TMath.h
+++ b/math/mathcore/inc/TMath.h
@@ -577,62 +577,12 @@ struct Limits {
 // Trig and other functions
 
 #include <float.h>
+#include <math.h>
 
 #if defined(R__WIN32) && !defined(__CINT__)
 #   ifndef finite
 #      define finite _finite
 #   endif
-#endif
-#if defined(R__AIX) || defined(R__SOLARIS_CC50) || \
-    defined(R__HPUX11) || defined(R__GLIBC) || \
-    (defined(R__MACOSX) )
-// math functions are defined inline so we have to include them here
-#   include <math.h>
-#   ifdef R__SOLARIS_CC50
-       extern "C" { int finite(double); }
-#   endif
-// #   if defined(R__GLIBC) && defined(__STRICT_ANSI__)
-// #      ifndef finite
-// #         define finite __finite
-// #      endif
-// #      ifndef isnan
-// #         define isnan  __isnan
-// #      endif
-// #   endif
-#else
-// don't want to include complete <math.h>
-extern "C" {
-   extern double sin(double);
-   extern double cos(double);
-   extern double tan(double);
-   extern double sinh(double);
-   extern double cosh(double);
-   extern double tanh(double);
-   extern double asin(double);
-   extern double acos(double);
-   extern double atan(double);
-   extern double atan2(double, double);
-   extern double sqrt(double);
-   extern double exp(double);
-   extern double pow(double, double);
-   extern double log(double);
-   extern double log10(double);
-#ifndef R__WIN32
-#   if !defined(finite)
-       extern int finite(double);
-#   endif
-#   if !defined(isnan)
-       extern int isnan(double);
-#   endif
-   extern double ldexp(double, int);
-   extern double ceil(double);
-   extern double floor(double);
-#else
-   _CRTIMP double ldexp(double, int);
-   _CRTIMP double ceil(double);
-   _CRTIMP double floor(double);
-#endif
-}
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/net/rpdutils/src/daemon.cxx
+++ b/net/rpdutils/src/daemon.cxx
@@ -24,8 +24,7 @@
 #include <signal.h>
 #include <sys/stat.h>
 #include <sys/param.h>
-#if defined(__sun) || defined(__sgi)
-#  include <fcntl.h>
+#include <fcntl.h>
 #endif
 
 #if defined(__linux__) && !defined(linux)

--- a/net/rpdutils/src/daemon.cxx
+++ b/net/rpdutils/src/daemon.cxx
@@ -43,15 +43,6 @@
 #   define NOFILE 0
 #endif
 
-#if defined(__hpux)
-#define USE_SIGCHLD
-#endif
-
-#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
-#define USE_SIGCHLD
-#define SIGCLD SIGCHLD
-#endif
-
 #if defined(linux) || defined(__hpux) || defined(__sun) || defined(__sgi) || \
     defined(_AIX) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
     defined(__APPLE__) || defined(__MACH__) || \
@@ -68,23 +59,16 @@ namespace ROOT {
 
 extern ErrorHandler_t gErrSys;
 
-#if defined(USE_SIGCHLD)
 ////////////////////////////////////////////////////////////////////////////////
 
 static void SigChild(int)
 {
-   int         pid;
-#if defined(__hpux) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
-    defined(__APPLE__)
+   int pid;
    int status;
-#else
-   union wait  status;
-#endif
 
    while ((pid = wait3(&status, WNOHANG, 0)) > 0)
       ;
 }
-#endif
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Detach a daemon process from login session context.
@@ -207,15 +191,7 @@ out:
    // and execute the wait3() system call.
 
    if (ignsigcld) {
-#ifdef USE_SIGCHLD
-      signal(SIGCLD, SigChild);
-#else
-#if defined(__sun)
-      sigignore(SIGCHLD);
-#else
-      signal(SIGCLD, SIG_IGN);
-#endif
-#endif
+      signal(SIGCHLD, SigChild);
    }
 }
 

--- a/net/rpdutils/src/daemon.cxx
+++ b/net/rpdutils/src/daemon.cxx
@@ -25,7 +25,6 @@
 #include <sys/stat.h>
 #include <sys/param.h>
 #include <fcntl.h>
-#endif
 
 #if defined(__linux__) && !defined(linux)
 # define linux

--- a/net/rpdutils/src/net.cxx
+++ b/net/rpdutils/src/net.cxx
@@ -31,17 +31,6 @@
 #include <netdb.h>
 #include <errno.h>
 
-#if (defined(R__AIX) && !defined(_AIX43)) || \
-    (defined(R__SUNGCC3) && !defined(__arch64__))
-#   define USE_SIZE_T
-#elif defined(R__GLIBC) || defined(R__FBSD) || \
-     (defined(R__SUNGCC3) && defined(__arch64__)) || \
-     defined(R__OBSD) || defined(MAC_OS_X_VERSION_10_4) || \
-     (defined(R__AIX) && defined(_AIX43)) || \
-     (defined(R__SOLARIS) && defined(_SOCKLEN_T))
-#   define USE_SOCKLEN_T
-#endif
-
 #include "rpdp.h"
 #include "rpderr.h"
 
@@ -357,13 +346,7 @@ int NetRecv(char *msg, int max)
 
 int NetOpen(int inetdflag, EService service)
 {
-#if defined(USE_SIZE_T)
-   size_t clilen = sizeof(gTcpCliAddr);
-#elif defined(USE_SOCKLEN_T)
    socklen_t clilen = sizeof(gTcpCliAddr);
-#else
-   int clilen = sizeof(gTcpCliAddr);
-#endif
 
    if (inetdflag) {
 
@@ -588,13 +571,7 @@ void NetSetOptions(EService serv, int sock, int tcpwindowsize)
          ErrorInfo("NetSetOptions: set SO_RCVBUF %d", val);
 
    if (gDebug > 0) {
-#if defined(USE_SIZE_T)
-      size_t optlen = sizeof(val);
-#elif defined(USE_SOCKLEN_T)
       socklen_t optlen = sizeof(val);
-#else
-      int optlen = sizeof(val);
-#endif
       if (serv == kROOTD) {
          getsockopt(sock, IPPROTO_TCP, TCP_NODELAY, (char*)&val, &optlen);
          ErrorInfo("NetSetOptions: get TCP_NODELAY: %d", val);

--- a/net/rpdutils/src/netpar.cxx
+++ b/net/rpdutils/src/netpar.cxx
@@ -37,17 +37,6 @@
 #include <strings.h>
 #endif
 
-#if (defined(R__AIX) && !defined(_AIX43)) || \
-    (defined(R__SUNGCC3) && !defined(__arch64__))
-#   define USE_SIZE_T
-#elif defined(R__GLIBC) || defined(R__FBSD) || \
-      (defined(R__SUNGCC3) && defined(__arch64__)) || \
-      defined(R__OBSD) || defined(MAC_OS_X_VERSION_10_4) || \
-      (defined(R__AIX) && defined(_AIX43)) || \
-      (defined(R__SOLARIS) && defined(_SOCKLEN_T))
-#   define USE_SOCKLEN_T
-#endif
-
 #include "rpdp.h"
 
 extern int gDebug;
@@ -201,13 +190,7 @@ int NetParOpen(int port, int size)
    struct sockaddr_in remote_addr;
    memset(&remote_addr, 0, sizeof(remote_addr));
 
-#if defined(USE_SIZE_T)
-   size_t remlen = sizeof(remote_addr);
-#elif defined(USE_SOCKLEN_T)
    socklen_t remlen = sizeof(remote_addr);
-#else
-   int remlen = sizeof(remote_addr);
-#endif
 
    if (!getpeername(NetGetSockFd(), (struct sockaddr *)&remote_addr, &remlen)) {
       remote_addr.sin_family = AF_INET;

